incentivization.md

```diff
@@ -0,0 +1,342 @@
+
+- the server responds quickly;
+- all the messages in the response are relevant;
+- the response contains only relevant messages.
+
+# Waku Store incentivization MVP
+
+In this section, we aim to define the simplest viable i13n modification to the Store protocol.
+
+We propose to add the following aspects to the protocol:
+1. price offer;
+2. proof of payment;
+3. reputation accounting.
+
+### Price offer
+
+After the client sends a `HistoryQuery` to the server:
+1. The server internally calculates the offer price  and sends it to the client.
+2. If the client agrees, it pays and sends a proof of payment to the server.
+3. If the client does not agree, it sends a rejection message to the server.
+4. If the server receives a valid proof of payment before a certain timeout, it sends the response to the client.
+5. If the server receives a rejection message, or receives no message before a timeout, the server assumes that the client has rejected the offer.
+
+Potential issues:
+- The client overwhelms a server with requests but doesn't proceed with payment. Countermeasure: ignore requests from the same client if they come too often.
+- The server and the client have no means to negotiate the price - see a later section on price negotiation.
+
+### Proof of payment
+
+If the client agrees to the price, it sends a _proof of payment_ to the server.
+The nature of such proof depends on the means of payment.
+Assuming the payment takes place on a blockchain, it could simply be a transaction hash (`txid`).
+
+It's unclear whether we need to ensure that a given `txid` is linked to a particular request.
+Including request ID into the payment (a-la "memo field") threatens privacy.
+Not including it could lead to the server's confusion regarding which received payments correspond to which requests.
+
+#### Who pays first?
+
+We have to make a design decision: who pays first?
+Our options are:
+1. the client pays first;
+2. the client pays after the fac;
+3. the client pays partly upfront and partly after the fact;
+4. a third party (escrow) ensures atomicity (it may be a centralized trusted third party or a semi-trusted entity like a smart contract).
+
+Our design considerations are:
+- the MVP protocol should be simple;
+- servers are more "permanent" entities and are more likely to have a long-lived identities;
+- it is more important to protect the clients's privacy than the server's privacy: a client knows what server it queries, while the server ideally shouldn't know who the client is.
+
+With that in mind, we suggest that the client pays first.
+It is simpler than splitting the payment, which would involve a) two payments, and b) negotiating the split.
+It is also simpler than a trusted third party (the centralized flavor of which we want to avoid).
+
+Comparing to "client pays after the fact", we observe that there is a balance between risk and privacy.
+If the server "pays first", it assumes risk, which should be decreased or paid for.
```

Comment:
This whole part is strange - it feels (like @mattyTokenomics points out below) like there is a mixup of server and client in this paragraph

Reply:
I'll rewrite this in simpler terms. The key idea is: the client pays first because it's simple and arguably more privacy preserving for the client (although the latter statement may require its own investigation).